src/jennifer/adapter/adapter_registry.cr

```diff
@@ -0,0 +1,51 @@
+module Jennifer
+  module Adapter
+    class AdapterRegistry
+
+      @@adapter_classes = {} of String => Base.class
+
+      @@adapter_instances = {} of Symbol => Jennifer::Adapter::Base
+
+      # Register an adapter class with a given name.
+      #
+      def self.register_adapter(name : String, adapter_class : Jennifer::Adapter::Base.class)
+        @@adapter_classes[name] = adapter_class
+      end
+
+      # temporary and will be refactored away when adapters can co-exist
+      {% if env("DB") == "mysql" %}
```

Comment:
in this case firstly way how adapters work should be refactored before merging this (we can't depends on env variable)

Reply:
I'll move the registration out of here and into the compile time selected adapter logic. that way it can :ship: before the adapter refactor